source/vibe/core/core.d

```diff
@@ -1169,12 +1176,14 @@ private class VibeDriverCore : DriverCore {
 			debug if (s_taskEventCallback) s_taskEventCallback(TaskEvent.resume, task);
 			// leave fiber.m_exception untouched, so that it gets thrown on the next yieldForEvent call
 		} else {
-			assert(!s_eventLoopRunning, "Event processing outside of a fiber should only happen before the event loop is running!?");
-			m_eventException = null;
-			try if (auto err = getEventDriver().runEventLoopOnce()) {
-				logError("Error running event loop: %d", err);
-				assert(err != 1, "No events registered, exiting event loop.");
-				assert(false, "Error waiting for events.");
+			try {
+				assert(!s_eventLoopRunning, "Event processing outside of a fiber should only happen before the event loop is running!?");
+				m_eventException = null;
+				if (auto err = getEventDriver().runEventLoopOnce()) {
```

Comment:
Why the change ? `assert` and a member assignement should be `nothrow`, aren't they ?